STICS' write.config using vectors of parameters to overwrite

qdbell opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!--- Please select appropriate Priority, Status,and Type labels-->
<!--- If you do not have permission to select labels please state which labels you would like -->

## Description
Modified the usage of SticsOnR:set_param_xml() to utilise the vector functionality where many parameters can be passed to write them at once. This entailed removing the individual parameter writing calls and adding a conversion function pecan2stics() that takes pecan parameter names and converts them to the corresponding stics parameter names, along with any necessary unit conversions.

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
This change should speed up the use of PEcAn.STICS as there were many individual set_param_xml calls so it was slow to write configs.

## Review Time Estimate
<!---When do you want your code reviewed by?-->
- [ ] Immediately
- [ ] Within one week
- [x] When possible

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue) <!-- please add issue number -->
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My change requires a change to the documentation.
- [x] My name is in the list of CITATION.cff
- [x] I agree that PEcAn Project may distribute my contribution under any or all of
	- the same license as the existing code,
	- and/or the BSD 3-clause license.
- [x] I have updated the CHANGELOG.md.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

<!--this template is from https://www.talater.com/open-source-templates/#/page/99-->

qdbell committed:
Switched from individual parameter writes to using the vector functionality of SticsRFiles::set_param_xml.

qdbell committed:
Added Quentin Bell and write.config.STICS change

istfer commented on CHANGELOG.md:
```diff
@@ -22,6 +22,7 @@ For more information about this file see also [Keep a Changelog](http://keepacha
     * Modules `PEcAn.allometry`, `PEcAn.assim.batch`, `PEcAn.data.mining`, `PEcAn.emulator`, `PEcAn.MA`, `PEcAn.photosynthesis`, `PEcAn.priors`, and `PEcAn.RTM`.
 - Renamed master branch to main
 - `PEcAn.all::pecan_version()` now reports commit hashes as well as version numbers for each installed package.
+- `write.conmfig.STICS()` now modifies parameters with vectors rather than individually.
```
```suggestion
- `write.config.STICS()` now modifies parameters with vectors rather than individually.
```

qdbell committed:
Update write.config.STICS.R with STICS filenames

Added which STICS file the parameter is found in to the pecan2stics lookup table.

dlebauer committed:
Merge branch 'develop' into stics_vector_param

dlebauer committed:
Merge branch 'develop' into stics_vector_param

infotroph committed:
Update CHANGELOG.md

Co-authored-by: Istem Fer <fer.istem@gmail.com>

infotroph committed:
Merge branch 'develop' into stics_vector_param

infotroph commented on models/stics/R/write.config.STICS.R:
```diff
@@ -1274,62 +437,69 @@ write.config.STICS <- function(defaults, trait.values, settings, run.id) {
 
-  # I guess not important what this is called as long as it's consistent in usms
-  SticsRFiles::set_soil_txt(file = sols_file, param="typsol", value=paste0("sol", str_ns))
+  soils_df <- data.frame(soil_name = paste0("sol", str_ns))
```
@qdbell @istfer Heads up that I resolved a merge conflict (as commit 1f98e701) here  -- the changes didn't actually overlap, but it was a good reminder that I should ask if my changes in #3428  work with your filename logic.

infotroph committed:
Merge branch 'develop' into stics_vector_param

infotroph committed:
Merge branch 'develop' into stics_vector_param

infotroph commented on CHANGELOG.md:
```diff
@@ -79,6 +79,7 @@ section for the next release.
     * Modules `PEcAn.allometry`, `PEcAn.assim.batch`, `PEcAn.data.mining`, `PEcAn.emulator`, `PEcAn.MA`, `PEcAn.photosynthesis`, `PEcAn.priors`, and `PEcAn.RTM`.
 - Renamed master branch to main
 - `PEcAn.all::pecan_version()` now reports commit hashes as well as version numbers for each installed package.
+- `write.config.STICS()` now modifies parameters with vectors rather than individually.
```
The changelog has grown above this line and it's now in the section for an old release. Please move up to the "unreleased" section at the top of the file

qdbell committed:
Update CHANGELOG.md

istfer commented:
sorry @infotroph I missed previous notifications on this, is there anything else remaining for us to check? thank you for keeping it updated

infotroph commented:
@istfer Looks like the current merge blocker is the package checks complaining about an unexported function:

```
  checking dependencies in R code ... WARNING
  '::' or ':::' import not declared from: ‘tibble’
  Unexported object imported by a ':::' call: ‘SticsRFiles:::gen_sol_xsl_file’
    See the note in ?`:::` about the use of this operator.
```

I only checked briefly, but it looks like you may have been using `gen_sol_xsl_file` as a workaround for a limitation in `convert_xml2txt`. If so, does using SticsRFiles >= 1.6.0 help? I see its NEWS [includes](https://github.com/SticsRPacks/SticsRFiles/blob/main/NEWS.md#sticsrfiles-160-2025-06-03) "convert_xml2txt allows to convert sols.xml file to text files".

Alternately, if there's not a clean way to avoid using `gen_sol_xsl_file` then  I suggest ignoring the message by pasting the check message lines I quoted above into Rcheck_reference.log replacing current lines 83-85.

qdbell committed:
Update write.config.STICS.R

Remove step of soil file conversion due to update in SticsRFiles.

infotroph committed:
Merge branch 'develop' into stics_vector_param

infotroph commented on models/stics/R/write.config.STICS.R:
```diff
@@ -1752,5 +922,193 @@ write.config.STICS <- function(defaults, trait.values, settings, run.id) {
+pecan2stics <- function(trait.values){
+  
+  # TODO :match all stics and pecan names
+  vartable <- tibble::tribble(
```
you can avoid needing to declare a new import by using the version in dplyr:
```suggestion
  vartable <- dplyr::tribble(
```

infotroph committed:
Update models/stics/R/write.config.STICS.R
